mlrun/launcher/base.py

```diff
@@ -248,6 +248,10 @@ def _enrich_run(
         # for other runtimes we need to convert the handler to a string
         if run.spec.handler and runtime.kind not in ["handler", "dask"]:
             run.spec.handler = run.spec.handler_name
+            function_kind = getattr(runtime.spec, "function_kind", None)
+            mlrun.utils.helpers.validate_handler_name(
+                function_kind=function_kind, handler=run.spec.handler
+            )
```

Comment:
at this point the function is already deployed and you fail validation when running it. 
why not also validate `runtime.spec.default_handler` in the deploy? can be validated in `mlrun.runtimes.nuclio.function.RemoteRuntime.pre_deploy_validation`

Reply:
But the handler name/function is being passed through the run

Reply:
I missed that you added validation on `code_to_function` which is a good start but left a comment there for what is missing